metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/data_classes.py

```diff
@@ -286,11 +286,13 @@ class CreatedFrom(Enum):
     id: str
     title: str
     embedUrl: str
-    dataset: Optional["PowerBIDataset"]
     dataset_id: Optional[str]
-    report: Optional[Report]
+    report_id: Optional[str]
     createdFrom: CreatedFrom
 
+    dataset: Optional["PowerBIDataset"]
+    report: Optional[Report]
```

Comment:
are report_id and report set at the same time? if not, it might be worth putting a comment that says the latter is filled in at a later time

Reply:
ids are set on a first traversal and objects in a later one
I will add a comment